sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundle.java

```diff
@@ -0,0 +1,289 @@
+   * Spring usually provides {@link org.springframework.data.convert.Jsr310Converters} that are not
+   * used here intentionally.
+   */
+  private static final Set<Converter<?, ?>> DEFAULT_CONVERTERS =
+      new LinkedHashSet<>(
+          List.of(
+              // LocalDate
+              new LocalDateToStringConverter(),
+              new StringToLocalDateConverter(),
+              // ZonedDateTime
+              new DateToZonedDateTimeConverter(),
+              new StringToZonedDateTimeConverter(),
+              new ZonedDateTimeToDateConverter()));
+
+  private final Function<C, SpringDataMongoConfiguration> configurationProvider;
+
+  private SpringDataMongoConfiguration config;
+
+  private MongoClient mongoClient;
+
+  private MongoOperations mongoOperations;
+
+  private final Set<Converter<?, ?>> customConverters = new LinkedHashSet<>();
+
+  private boolean autoIndexCreation = true;
+
+  /**
+   * Database as defined by the {@link SpringDataMongoConfiguration#getConnectionString()} or {@link
+   * SpringDataMongoConfiguration#getDatabase()}
+   */
+  private String database;
+
+  public SpringDataMongoBundle(SpringDataMongoConfigurationProvider<C> configurationProvider) {
+    this.configurationProvider = configurationProvider;
+  }
+
+  public static InitialBuilder builder() {
+    return new Builder<>();
+  }
+
+  @Override
+  public void initialize(Bootstrap<?> bootstrap) {
+    // nothing to do ATM
+  }
+
+  @Override
+  public void run(C configuration, Environment environment) {
+    this.config = configurationProvider.apply(configuration);
+
+    String connectionString;
+    if (StringUtils.isNotBlank(config.getConnectionString())) {
+      connectionString = config.getConnectionString();
+    } else {
+      connectionString =
+          String.format(
+              "mongodb://%s:%s@%s/%s",
+              config.getUsername(), config.getPassword(), config.getHosts(), config.getDatabase());
+      if (StringUtils.isNotBlank(config.getOptions())) {
+        connectionString += "?" + config.getOptions();
+      }
+    }
+
+    var cs = new ConnectionString(connectionString);
+    this.database = cs.getDatabase();
+    mongoClient = MongoClients.create(cs);
+
+    registerHealthCheck(environment.healthChecks(), mongoClient, this.database);
+  }
+
+  /**
+   * registers a health check for the mongo database
+   *
+   * @param healthCheckRegistry registry where to register health checks
+   * @param database database name that is used within health check
+   */
+  private void registerHealthCheck(
+      HealthCheckRegistry healthCheckRegistry, MongoClient mongoClient, String database) {
+    healthCheckRegistry.register(
+        "mongo", new MongoHealthCheck(mongoClient.getDatabase((database))));
+  }
+
+  /**
+   * @return the {@link MongoOperations} that can be used to interact with your MongoDB instance
+   */
+  public MongoOperations getMongoOperations() {
```

Comment:
I would ensure, that we do not initialize the `mongoOperations` twice:

```suggestion
  public synchronized MongoOperations getMongoOperations() {
```

Reply:
Better safe than sorry. Although I don't expect the method to be invoked many times. It should be used on running your application when you initialize your services. Application's will usually only have one instance.